charts/tidb-operator/templates/initializer-job.yaml

```diff
@@ -0,0 +1,51 @@
+{{ $a := split ":" .Values.operatorImage }}
+  {{ $length := len $a }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  {{ if eq $length 2 }}
+  name: tidb-operator-initializer-{{ $a._1 }}
+  {{ else }}
+  name: tidb-operator-initializer-{{ $a._2 }}
+  {{ end }}
```

Comment:
> what is this for?

Adding tag version in `Initializer` Job name, if not , Operator would upgrade failed as Job already exists.

Reply:
Sounds weird, what if the operator release is upgraded to change the configuration but the operatorImage is not changed?

Reply:
I got it, this job reference the operatorImage in container spec. But a more serious problem is that this job may run multiple times accidentally.

Reply:
> I got it, this job reference the operatorImage in container spec. But a more serious problem is that this job may run multiple times accidentally.

yep, the job would be created and run each time Operator Upgraded, but it doesn't mean the resources needed to be created each times. Currently I found that the `manifests/create_cert.sh` would delete the older resources and create the new one and I reused it.

Reply:
I don't get your point, would the recreation cause a disruption of the webhook?

Reply:
At least, this job is not necessarily to run multiple times, seems like that we are working around helm.

Reply:
> At least, this job is not necessarily to run multiple times, seems like that we are working around helm.

I think this would be revised as `Initializer` wouldn't recreate resources if the resources exists.

Reply:
fix by helm hook.